### PR TITLE
Adding fx project

### PR DESCRIPTION
**Pull request by emashliles**
@gemcfadyen @jsuchy

I still need to fix the build.

### CODE REVIEW

File: TTTFx/src/main/java/BoardController.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        }, 1000, 1000);
+    }
+
+    @FXML
+    protected void spaceSelected(ActionEvent actionEvent) throws IOException {
+        setInfoText("");
+
+        Button space = (Button)actionEvent.getSource();
+
+        if (checkForGameOver(space)) return;
```
Comment (gemcfadyen) [inline TTTFx/src/main/java/BoardController.java:79]: I would recommend you always use brackets for 'if' loops, even if the body is  just one line.

File: TTTFx/src/main/java/BoardController.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+        turnHandler.getPlayerTurn(spaceString, player1, player2, game.getCurrentPlayer());
+        turnHandler.doTurn(game, board);
+
+        setGameOverMessage(turnHandler.lastPlayerToMove());
+        updateBoard();
+    }
+
+    private void doTurns() {
+        if (game.isOver(board) == GameState.NoWinner) {
```
Comment (gemcfadyen) [inline TTTFx/src/main/java/BoardController.java:91]: The method `isOver` implies a boolean would be returned, maybe it should be called `getState` or similar?

File: TTTFx/src/main/java/ComputerVComputerTurnHandler.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+public class ComputerVComputerTurnHandler implements TurnHandler{
```
Comment (gemcfadyen) [inline TTTFx/src/main/java/ComputerVComputerTurnHandler.java:1]: missing space before `{`

File: TTTFx/src/main/java/ComputerVHumanTurnHandler.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
+public class ComputerVHumanTurnHandler implements TurnHandler{
```
Comment (gemcfadyen) [inline TTTFx/src/main/java/ComputerVHumanTurnHandler.java:1]: Missing space before `{`

File: TTTFx/src/main/java/HumanVComputerTurnHandler.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+public class HumanVComputerTurnHandler implements TurnHandler{
```
Comment (gemcfadyen) [inline TTTFx/src/main/java/HumanVComputerTurnHandler.java:1]: Missing space before `{`

File: TTTFx/src/main/java/HumanVHumanTurnHandler.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
+public class HumanVHumanTurnHandler implements TurnHandler {
+
+    private PlayerMarkers lastPlayerMoved;
+
+    public void getPlayerTurn(String spaceString, Player player1, Player player2, PlayerMarkers currentPlayer) {
+        HumanFxPlayer player1Human = (HumanFxPlayer) player1;
+        HumanFxPlayer player2Human = (HumanFxPlayer) player2;
+
+        if(currentPlayer == PlayerMarkers.X) {
```
Comment (gemcfadyen) [inline TTTFx/src/main/java/HumanVHumanTurnHandler.java:9]: Can you think of a way to remove the `if` statement here?

File: TTTFx/src/test/java/ComputerVHumanTests.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+        boardSceneCreator.doComputerTurnFirst(controller);
+        boardSceneCreator.showScene(parent);
+    }
+
+    @Test
+    public void computerMoveIsDisplayedFirst() {
+        verifyThat("#Space_1", hasText("X"));
+    }
+
```
Comment (gemcfadyen) [inline TTTFx/src/test/java/ComputerVHumanTests.java:28]: surplus line here

**Review by gemcfadyen — COMMENTED** [review]
I think you achieved a lot this sprint - well done! General comments - Have a think about what would need to change if we decided to implement a second gui using swing for example. (i.e: evaluate how tied to JavaFX is your implementation at present?)

File: TTTFx/src/main/java/ComputerVComputerTurnHandler.java (hunk 3 above)
Comment (emashliles) [reply TTTFx/src/main/java/ComputerVComputerTurnHandler.java:1]: Updated in https://github.com/emashliles/java-tic-tac-toe/commit/9f4bdf80e166a3635ae5b36a07eadbf22d102c9d

File: TTTFx/src/main/java/ComputerVHumanTurnHandler.java (hunk 4 above)
Comment (emashliles) [reply TTTFx/src/main/java/ComputerVHumanTurnHandler.java:1]: Updated in https://github.com/emashliles/java-tic-tac-toe/commit/9f4bdf80e166a3635ae5b36a07eadbf22d102c9d

File: TTTFx/src/main/java/HumanVComputerTurnHandler.java (hunk 5 above)
Comment (emashliles) [reply TTTFx/src/main/java/HumanVComputerTurnHandler.java:1]: Updated in https://github.com/emashliles/java-tic-tac-toe/commit/9f4bdf80e166a3635ae5b36a07eadbf22d102c9d

**Pull request merged**